File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next    
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        vals = []
        node = head
        while node:
            vals.append(node.val)
            node = node.next
        
        left = 0
        right = len(vals) - 1
        
        while left <= right:
            if vals[left] != vals[right]:
                return False
            left += 1
            right -= 1
        
        return True

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */

var isPalindrome = function(head) {
    let vals = [];
    let node = head;
    
    while (node) {
        vals.push(node.val);
        node = node.next;
    }
    
    let left = 0;
    let right = vals.length - 1;
    
    while (left <= right) {
        if (vals[left] !== vals[right]) {
            return false;
        }
        left++;
        right--;
    }
    
    return true;
    
};